source/Calamari/Kubernetes/Integration/HelmCli.cs

```diff
@@ -130,5 +146,60 @@ void ChmodExecutable()
                                                                      ExecutableLocation));
             }
         }
+
+        CommandResult ExecuteCommandViaScript(IEnumerable<string> arguments)
+        {
+            var syntax = ScriptSyntaxHelper.GetPreferredScriptSyntaxForEnvironment();
+            var command = BuildHelmScriptCommand(arguments, syntax);
+            var fileName = SyntaxSpecificFileName(syntax);
+            var scriptExecutor = GetScriptExecutor(syntax);
+            
+            using (new TemporaryFile(fileName))
+            {
+                fileSystem.OverwriteFile(fileName, command);
+                
+                return scriptExecutor.Execute(new Script(fileName), variables, commandLineRunner, environmentVars);
+            }
+        }
+
+        string BuildHelmScriptCommand(IEnumerable<string> arguments, ScriptSyntax syntax)
+        {
+            var cmd = new List<string>();
+            
+            SetExecutable(cmd, syntax);
+            cmd.AddRange(arguments.Where(s => !string.IsNullOrWhiteSpace(s)));
+
+            var command = string.Join(" ", cmd);
+            log.Verbose(command);
+            return command;
+        }
+        
+        void SetExecutable(List<string> commandBuilder, ScriptSyntax syntax)
+        {
+            if (isCustomExecutable)
+            {
+                commandBuilder.Add(syntax == ScriptSyntax.PowerShell ? ". " : $"chmod +x \"{ExecutableLocation}\";");
```

Comment:
A lot of this is copied over from https://github.com/OctopusDeploy/Calamari/blob/main/source/Calamari/Kubernetes/Conventions/HelmUpgradeConvention.cs